addon/components/polaris-resource-list.js

```diff
@@ -1,34 +1,797 @@
+
+      if (onSelectionChange) {
+        onSelectionChange(newlySelectedItems);
+      }
+    },
+
+    handleSelectMode(selectMode) {
+      let onSelectionChange = this.get('onSelectionChange');
+      this.set('selectMode', selectMode);
+      if (!selectMode && onSelectionChange) {
+        onSelectionChange([]);
+      }
+    },
+
+    handleToggleAll() {
+      let {
+        onSelectionChange,
+        selectedItems,
+        items,
+        idForItem,
+      } = this.getProperties(
+        'onSelectionChange',
+        'selectedItems',
+        'items',
+        'idForItem'
+      );
+      idForItem = idForItem || defaultIdForItem;
+
+      let newlySelectedItems = [];
+
+      if (
+        (Array.isArray(selectedItems) &&
+          selectedItems.length === items.length) ||
+        selectedItems === SELECT_ALL_ITEMS
+      ) {
+        newlySelectedItems = [];
+      } else {
+        newlySelectedItems = items.map((item, index) => {
+          let id = idForItem(item, index);
+          return id;
+        });
+      }
+
+      if (newlySelectedItems.length === 0 && !isSmallScreen()) {
+        this.handleSelectMode(false);
+      } else if (newlySelectedItems.length > 0) {
+        this.handleSelectMode(true);
+      }
+
+      if (onSelectionChange) {
+        onSelectionChange(newlySelectedItems);
+      }
+    },
+
+    setListNode() {
+      let listNode =
+        document.querySelector(
+          `#${this.get('wrapperId')} ul.Polaris-ResourceList`
+        ) || null;
+      this.set('listNode', listNode);
+    },
+
+    init() {
+      this._super(...arguments);
+
+      this.set('defaultResourceName', {
+        singular: 'item',
+        plural: 'items',
+      });
+    },
+
+    didReceiveAttrs() {
+      this._super(...arguments);
+
+      assert(
+        'ember-polaris::polaris-resource-list - itemComponent must be a component name or definition',
+        this.get('itemComponent')
+      );
+
+      // This logic is in the React implementation's
+      // `componentWillReceiveProps` hook.
+      let { selectedItems, previousSelectedItems } = this.getProperties(
+        'selectedItems',
+        'previousSelectedItems'
+      );
+
+      if (
+        previousSelectedItems &&
+        previousSelectedItems.length > 0 &&
+        (!selectedItems || selectedItems.length === 0) &&
+        !isSmallScreen()
+      ) {
+        this.set('selectMode', false);
+      }
+
+      // This logic is in the React implementation's
+      // `componentDidUpdate` hook.
+      let { loading, previousLoading } = this.getProperties(
+        'loading',
+        'previousLoading'
+      );
+
+      if (loading && !previousLoading) {
+        this.setLoadingPosition();
+      }
+
+      this.setProperties({
+        previousSelectedItems: selectedItems,
+        previousLoading: loading,
+      });
+    },
+
+    didInsertElement() {
+      this._super(...arguments);
+
+      this.addEventListener(window, 'resize', this.handleResize);
```

Comment:
It looks like [react](https://github.com/Shopify/polaris-react/blob/v2.11.0/src/components/ResourceList/ResourceList.tsx#L353) only has this listener in the DOM when `selectable` is true, would it be worth it to add something like:

```javascript
debouncedHandleResize() {
  if (!this.get('selectable') {
    return;
  }
 // ...
}
```

at the beginning of `debouncedHandleResize`? I'm fine with leaving it though, doesn't seem to be any harm if this gets called when `selectable` is false.